lib/container.ts

```diff
@@ -79,310 +70,281 @@ export class Container extends (EventEmitter as {
 	// tslint:disable-next-line
 	new (): ContainerEventEmitter;
 }) {
-	private dockerfile: Dockerfile;
-
-	public constructor(
-		public readonly dockerfileContent: string,
-		public readonly hostContextPath: string,
+	private constructor(
+		private buildContext: string,
+		private docker: Docker,
 		public containerId: string,
-		public readonly docker: Dockerode,
 	) {
 		super();
-		this.dockerfile = new Dockerfile(dockerfileContent);
 	}
 
-	public actionsNeeded(files: FileUpdates): DockerfileActionGroup[] {
-		// The Dockerfile class itself doesn't care whether a file has been added, changed or deleted,
-		// it simply returns the commands which need to happen based on the paths in the COPY directives.
-		return this.dockerfile.getActionGroupsFromChangedFiles(files.affected());
+	public static fromContainerId(
```

Comment:
NIT: this only passes arguments to the constructor and requires more than a container id.